test/Mono.Linker.Tests.Cases/DataFlow/CompilerGeneratedCodeDataflow.cs

```diff
@@ -0,0 +1,540 @@
+			[ExpectedWarning ("IL2072", nameof (GetWithPublicFields), nameof (DataFlowTypeExtensions.RequiresPublicMethods), CompilerGeneratedCode = true)]
+			static async void NoFlowAcrossAwait ()
+			{
+				Type t = GetWithPublicMethods ();
+				t.RequiresPublicMethods ();
+				await MethodAsync ();
+				t = GetWithPublicFields ();
+				t.RequiresPublicFields ();
+			}
+
+			[ExpectedWarning ("IL2067", "publicMethodsType", nameof (DataFlowTypeExtensions.RequiresAll), CompilerGeneratedCode = true)]
+			static async void FlowParameterAcrossAwait ([DynamicallyAccessedMembers (DynamicallyAccessedMemberTypes.PublicMethods)] Type publicMethodsType = null)
+			{
+				await MethodAsync ();
+				publicMethodsType.RequiresAll ();
+			}
+
+			public static void Test ()
+			{
+				FlowAcrossAwait ();
+				NoFlowAcrossAwait ();
+				FlowParameterAcrossAwait ();
+			}
+		}
+
+		class AsyncIterator
+		{
+			[ExpectedWarning ("IL2072", nameof (GetWithPublicMethods), nameof (DataFlowTypeExtensions.RequiresAll), CompilerGeneratedCode = true,
+				ProducedBy = ProducedBy.Trimmer)]
+			static async IAsyncEnumerable<int> FlowAcrossAwaitAndYieldReturn ()
+			{
+				Type t = GetWithPublicMethods ();
+				await MethodAsync ();
+				yield return 0;
+				t.RequiresAll ();
+			}
+
+			public static void Test ()
+			{
+				FlowAcrossAwaitAndYieldReturn ();
+			}
+		}
+
+		class LocalFunction
+		{
+			static void WarningsInBody ()
+			{
+				[ExpectedWarning ("IL2072", nameof (GetWithPublicMethods), nameof (DataFlowTypeExtensions.RequiresAll),
+					ProducedBy = ProducedBy.Trimmer)]
+				static void LocalFunction ()
+				{
+					Type t = GetWithPublicMethods ();
+					t.RequiresAll ();
+				}
+
+				LocalFunction ();
+			}
+
+			static void WarningsInBodyUnused ()
+			{
+				// Trimmer doesn't warn because this is unused code.
+				static void LocalFunction ()
+				{
+					Type t = GetWithPublicMethods ();
+					t.RequiresAll ();
+				}
+			}
+
+			static void ReadCapturedVariable ()
+			{
+				Type t = GetWithPublicMethods ();
+				LocalFunction ();
+
+				[ExpectedWarning ("IL2072", nameof (GetWithPublicMethods), nameof (DataFlowTypeExtensions.RequiresAll),
+					ProducedBy = ProducedBy.Trimmer)]
+				void LocalFunction ()
+				{
+					t.RequiresAll ();
+				}
+			}
+
+			static void ReadMergedCapturedVariable (bool b = false)
+			{
+				Type t;
+				if (b) {
+					t = GetWithPublicMethods ();
+				} else {
+					t = GetWithPublicFields ();
+				}
+
+				LocalFunction ();
+
+				[ExpectedWarning ("IL2072", nameof (GetWithPublicMethods), nameof (DataFlowTypeExtensions.RequiresAll),
+					ProducedBy = ProducedBy.Trimmer)]
+				[ExpectedWarning ("IL2072", nameof (GetWithPublicFields), nameof (DataFlowTypeExtensions.RequiresAll),
+					ProducedBy = ProducedBy.Trimmer)]
+				void LocalFunction () => t.RequiresAll ();
+			}
+
+			static void ReadCapturedVariableInMultipleBranches (bool b = false)
```

Comment:
Could you add a test with something like

```C#
static void ReadCapturedVariableInMultipleBranches (bool b = false)
{
    Type t;
    if (b) {
        t = GetWithPublicMethods ();
        TRequiresMethods ();
    } else {
        t = GetWithPublicFields ();
        TRequiresFields ();
    }

    void TRequiresFields () => t.RequiresPublicFields ();
    void TRequiresMethods () => t.RequiresPublicMethods ();
}
```

My guess is that this shouldn't have to warn.